src/Game.js

```diff
@@ -0,0 +1,253 @@
+        let ballWinner = '';
+
+        if (winner === playerOne.name) {
+            ballWinner = { isPlayerOneBall: true, isPlayerTwoBall: false };
+            setWinnerBall(ballWinner);
+        } else if (winner === playerTwo.name) {
+            ballWinner = { isPlayerOneBall: false, isPlayerTwoBall: true };
+            setWinnerBall(ballWinner);
+        };
+
+        setRestartGame(true);
+        checkPointsForPlayer(winner);
+        setWinner('');
+
+        newFocusRef && newFocusRef.current && newFocusRef.current.focus();
+    };
+
+    const checkPointsForPlayer = (winner) => {
+        let result = '';
+
+        if (winner === playerOne.name) {
+            result = playerOne.score;
+            setPlayerOne({ ...playerOne, score: playerOne.score + 1 });
+        } else if (winner === playerTwo.name) {
+            result = playerTwo.score;
+            setPlayerTwo({ ...playerTwo, score: playerTwo.score + 1 });
+        };
+
+        return result;
+    };
+
+    const runBoardCheck = (board) => {
+        const winner = getWinner(board, { playerOneName: playerOne.name, playerTwoName: playerTwo.name, nInARow: nInARow });
+        winner && setWinner(winner);
+    };
+
+    const getMoveBallComponent = (props) => {
+        const { displayBoard, handleMoveDown, handleIsLastCellFull } = props;
+
+        return (
+            <MoveBall
+                displayBoard={displayBoard}
+                handleMoveDown={handleMoveDown}
+                handleIsLastCellFull={handleIsLastCellFull}
+                ref={moveBallRef}
+                shouldStop={winner}
+                shouldRestartGame={restartGame}
+                afterGameRestart={setRestartGame}
+                winnerBall={winnerBall}
+                rowSize={1}
+                colSize={boardSize.col}
+            />
+        );
+    };
+
+    const renderLoginForm = () => {
+        return (
+            <div>
+                {viewLoginForm
+                    ? <LoginForm
+                        setName={handleName}
+                        onClick={handleStartGame}
+                        isFirstInputValid={isValidUsername('playerOne', playerOne.name)}
+                        isSecondInputValid={isValidUsername('playerTwo', playerTwo.name)}
+                        shouldDisabledButton={playerOne.name.length < 3 || playerTwo.name.length < 3}
+                    />
+                    : <BoardSize
+                        setSize={handleBoardSize}
+                        onClick={handleLoginForm}
+                        nInARow={nInARow}
+                        isRowNumberValid={isValidNumber('rows', boardSize.row)}
+                        isColNumberValid={isValidNumber('columns', boardSize.col)}
+                        isNinARowValid={isValidNumber('nInARows', nInARow)}
+                        shouldDisabledButtonBoardSize={boardSize.row < 4 || boardSize.col < 4 || nInARow < 2}
+                    />
```

Comment:
What is the difference between LoginForm and BoardSize? I BoradSize should be renamed since it also renders a form. Also is it possible that LoginForm and BoardSize internally handle if the input is valid or not instead of passing it from props